build/xunit.runsettings

```diff
@@ -1,20 +1,44 @@
+  <RunConfiguration>
+    <!--
+      This setting controls the level of parallelism on process-level. Use 0 to enable the maximum process-level parallelism.
+
+      This setting determines the maximum number of test DLLs, or other test containers that can run in parallel. Each DLL runs in its own testhost process, and is isolated on the process level from the tests in other test DLLs. This setting doesn't force tests in each test DLL to run in parallel. Controlling the parallel execution within a DLL (on the thread-level) is up to the test framework such as MSTest, XUnit or NUnit.
+
+      The default value is 1, meaning that only one testhost runs at the same time. A special value 0 allows as many testhosts as you have logical processors (for example, 6, for a computer with 6 physical cores without multi-threading, or 12, for a computer with six physical cores with multi-threading).
+
+      The number of distinct DLLs in the run determines the actual number of testhosts started.
+    -->
+    <MaxCpuCount>4</MaxCpuCount>
+
+    <!-- Specify a Boolean value, which defines the exit code when no tests are discovered. If the value is true and no tests are discovered, a nonzero exit code is returned. Otherwise, zero is returned. -->
+    <TreatNoTestsAsError>true</TreatNoTestsAsError>
+
+    <EnvironmentVariables>
+      <!-- Changes the default timeout in VSTest for shutdown to 1000ms instead of 100ms which can cause intermittent failures in CI. -->
+      <VSTEST_TESTHOST_SHUTDOWN_TIMEOUT>1000</VSTEST_TESTHOST_SHUTDOWN_TIMEOUT>
+    </EnvironmentVariables>
+  </RunConfiguration>
+
+  <!-- https://xunit.net/docs/runsettings -->
+  <xUnit>
+    <!-- Set this value to true to include diagnostic information during test discovery and execution. Each runner has a unique method of presenting diagnostic messages. -->
+    <DiagnosticMessages>true</DiagnosticMessages>
+
+    <!-- Set this value to enable long-running (hung) test detection. When the runner is idle waiting for tests to finished, it will report that fact once the timeout has passed. Use a value of 0 to disable the feature, or a positive integer value to enable the feature (time in seconds). -->
+    <LongRunningTestSeconds>120</LongRunningTestSeconds>
+
+    <!-- Set this to override the maximum number of threads to be used when parallelizing tests within this assembly. Use a value of 0 or "default" to indicate that you would like the default behavior; use a value of -1 or "unlimited" to indicate that you do not wish to limit the number of threads used for parallelization. -->
+    <MaxParallelThreads>4</MaxParallelThreads>
+
+    <!-- Set this to true if this assembly is willing to participate in parallelization with other assemblies. Test runners can use this information to automatically enable parallelization across assemblies if all the assemblies agree to it. -->
+    <ParallelizeAssembly>false</ParallelizeAssembly>
```

Comment:
I think this setting nullifies the `MaxcpuCount (4)` property because as per the comment it looks like `MaxcpuCount specifies the number of distinct DLLs in the run determines the actual number of testhosts started.`. Given that the multiple assemblies are not running parallel then my understanding is that only one test host process is started. I guess we should change to true so that 4 DLLs can run in parallel.

Reply:
Since `MaxCpuCount` is set to `4` ([here](https://github.com/NuGet/NuGet.Client/pull/5874/files#diff-6b0fb5558d0f03f86a5e9cfee0a4f89ee36e018f308909358f9fe7b94e3bff34R13)), there will be 4 `vstest.console.exe` processes spawned.  I think it then launches each test assembly in those processes in parallel.  Then with `MaxParallelThreads` set to `4`, each test assembly runs with 4 threads (if there are enough CPUs).

This is really confusing since there are multiple test runners.  You can run tests under `xunit.runner.exe`, `vstest.console.exe`, etc.  I think the `ParallelizeAssembly` tells xunit specifically that it should run tests from multiple assemblies at the same time when running under certain test runners.  But since we're running the tests under `vstest.console.exe`, it seems to me that we should not be using xunit's parallelization.  I spent a lot of time read https://xunit.net/docs/runsettings but only experimenting with these numbers did I find a good balance between speed and reliability.  Do you know more about how all of this works?

Reply:
It looks like a similar issue has been discussed https://github.com/xunit/visualstudio.xunit/issues/191#issuecomment-1763975518. I think we are good here because it looks like xUnit parallelization is ignored by `dotnet test` command.